packages/react-toolbar/Spec.md

```diff
@@ -2,62 +2,139 @@
 
 ## Background
 
-_Description and use cases of this component_
+### Definition
+
+`Toolbar` is a surface that houses commands that operate on the content of the window, panel, or parent region it resides above. `Toolbar` is one of the most visible and recognizable way to surface commands, and can be an intuitive method for interacting with content on the page; however, if overloaded or poorly organized, they can be difficult to use and hide valuable commands from your user. `Toolbar` can also display a search box for finding content, hold simple commands as well as menus, or display the status of ongoing actions.
 
 ## Prior Art
 
-_Include background research done for this component_
+As a part of the spec definitions in Fluent UI, a research effort has been made through [Open UI](https://open-ui.org/). The current research proposal is available as an open source contribution undergoing review ([research proposal](https://github.com/openui/open-ui/pull/452))
+
+## Comparison of `@fluentui/react` and `@fluentui/react-northstar`
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+- All mentions of v7 or v8 == `@fluentui/react` ([docsite](https://developer.microsoft.com/en-us/fluentui#/))
+- All mentions of v0 == `@fluentui/react-northstar` ([docsite](https://fluentsite.z22.web.core.windows.net/))
 
-## Sample Code
+The main difference between `@fluentui/react`'s `CommandBar` and `@fluentui/react-northstar`'s `Toolbar` is the right group of commands present in `CommandBar`.
 
-_Provide some representative example code that uses the proposed API for the component_
+v0 `Toolbar` has support to children API with static components in `Toolbar` as `Toolbar.Button` etc...
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+The only layout variation is size differences, there are 2 sizes `medium` which is the default and `small`.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The `Toolbar` will implement a `children` based API and will leverage the use of `context` in the interaction and data flows of child components. To achieve `overflow` items once used inside `CommandBar` the component will leavarage from [priority-overflow](https://github.com/ling1726/priority-overflow)
 
-## Structure
+### Toolbar
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+The root level component serves as a simple container for all possible ToolbarComponents
+
+```typescript
+type ToolbarProps = {
+  /**
+   * Defines toolbar size
+   * @default medium
+   */
+  size?: 'small' | 'medium';
+};
+```
+
+### ToolbarButton
+
+It serves as an override on top of `Button` limiting the possible props only to `size`, `appearance`, `disabled` and `disabledFocusable`
```

Comment:
We wanted to be explicit about what we get here instead of using `Omit` and getting something unexpected that can latter be added to `ButtonProps`  and we are restricting these props because we want usage of components in the Toolbar component to be as standardized as much as possible with the design spec